docs/user-guide/kubectl-overview.md

```diff
@@ -33,160 +33,261 @@ Documentation for other releases can be found at
 
 ```
-kubectl [command] [TYPE] [NAME] [flags]
+kubectl [command] [TYPE] [NAME] -o=<output_format>
 ```
 
-This specifies:
+Depending on the `kubectl` operation, the following output formats are supported:
+
+Output format | Description
+--------------| -----------
+`-o=custom-columns=<spec>` | Print a table using a comma separated list of [custom columns](#custom-columns).
+`-o=custom-columns-file=<filename>` | Print a table using the [custom columns](#custom-columns) template in the `<filename>` file.
+`-o=json`     | Output a JSON formatted API object.
+`-o=jsonpath=<template>` | Print the fields defined in a [jsonpath](jsonpath.md) expression.
+`-o=jsonpath-file=<filename>` | Print the fields defined by the [jsonpath](jsonpath.md) expression in the `<filename>` file.
+`-o=name`     | Print only the resource name and nothing else. Supported only by [`kubectl label`](kubectl/kubectl_label.md).
+`-o=wide`     | Output in the default ps output format but include additional information like the node name.
+`-o=yaml`     | Output a YAML formatted API object.
+
+##### Example
+
+In this example, the following command outputs the details for a single pod in JSON format:
+
+`$ kubectl get -o json pod web-pod-13je7`
+
+Remember: See the [kubectl](kubectl/kubectl.md) reference documentation for details about which output format is supported by each command.
+
+#### Custom columns
+
+To define custom columns and output only the details that you want into a table, you can use the `custom-columns` option. You can choose to define the custom columns inline or use a template file: `-o=custom-columns=<spec>` or `-o=custom-columns-file=<filename>`.
 
-- `command` is a certain operation performed on a given resource(s), such as `create`, `get`, `describe`, `delete` etc.
-- `TYPE` is the type of resource(s). Both singular and plural forms are accepted. For example, `node(s)`, `namespace(s)`, `pod(s)`, `replicationcontroller(s)`, `service(s)` etc.
-- `NAME` is the name of resource(s). `TYPE NAME` can be specified as `TYPE name1 name2` or `TYPE/name1 TYPE/name2`. `TYPE NAME` can also be specified by one or more file arguments: `-f file1 -f file2 ...`, [use YAML rather than JSON](config-best-practices.md) since YAML tends to be more user-friendly for config.
-- `flags` are used to provide more control information when running a command. For example, you can use `-s` or `--server` to specify the address and port of the Kubernetes API server. Command line flags override their corresponding default values and environment variables. [Use short flags sparingly, only for the most frequently used options](../devel/kubectl-conventions.md).
+##### Examples
 
-Please use `kubectl help [command]` for detailed information about a command.
+ * Inline:
 
-Please refer to [kubectl](kubectl/kubectl.md) for a complete list of available commands and flags.
+      ```console
+      $ kubectl get pods <pod-name> -o=custom-columns=NAME:.metadata.name,RSRC:.metadata.resourceVersion
```

Comment:
Not sure how to use `-o=custom-columns` but this example gives me the following error: `output format "custom-columns=NAME:.metadata.name,RSRC:.metadata.resourceVersion" not recognized` cc @brendandburns